### PR TITLE
Vb 1936 add visits to the data returned on prisoner profile

### PR DESCRIPTION
**Pull request by dhirajshettymoj**
Include visit details when returning the prisoners profile

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/VisitSchedulerClient.kt [hunk 1]
```diff
@@ -27,115 +26,114 @@ import java.util.Optional
...
-  @Value("\${visit-scheduler.api.timeout:10s}") val apiTimeout: Duration,
 ) {
-  fun getVisitByReference(reference: String): VisitDto? {
+  fun getVisitByReference(reference: String): Mono<VisitDto> {
     return webClient.get()
       .uri("/visits/$reference")
       .accept(MediaType.APPLICATION_JSON)
       .retrieve()
-      .bodyToMono<VisitDto>().block(apiTimeout)
+      .bodyToMono()
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/VisitSchedulerClient.kt:35]: Not sure about these changes sorry, VisitSchedulerClient know about the client the receiver does not... use the POC this method should return a DTO not a frame work that is the concern of the client.

```
fun getVisitByReference(reference: String): CompletableFuture<VisitDto> {
    return webClient.get()
      .uri("/visits/$reference")
      .accept(MediaType.APPLICATION_JSON)
      .retrieve()
      .bodyToMono<VisitDto>().toFuture()
  }
```

**Review by AledWEvans — COMMENTED** [review]
have a look at comment's see what you think

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/PrisonerProfileService.kt [hunk 2]
```diff
@@ -1,52 +1,90 @@
...
+  @Value("\${prisoner.profile.future-visits.duration-in-months: P2M}") private val futureVisitsPeriod: Period,
 ) {
   companion object {
     val LOG: Logger = LoggerFactory.getLogger(this::class.java)
+    const val MAX_RECORDS: Int = 1000
+    const val PAGE: Int = 0
   }
 
   fun getPrisonerProfile(prisonId: String, prisonerId: String): PrisonerProfileDto? {
     val prisonerPromise = prisonerOffenderSearchClient.getPrisonerById(prisonerId)
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/PrisonerProfileService.kt:37]: I know I said a promise in another review but I suppose they are Futures..

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/VisitSchedulerClient.kt (hunk 1 above)
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/VisitSchedulerClient.kt]: As suggested I have updated the code to always return a DTO object except in one of the visitScheduler calls made by PrisonerProfile endpoint where it returns a Mono.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/PrisonerProfileService.kt (hunk 2 above)
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/PrisonerProfileService.kt]: Updated the variable names and also as discussed moved the logic from PrisonerProfileService to a new PrisonerProfileClient

**Review by AledWEvans — APPROVED** [review]
LGTM

**Pull request merged**